.github/workflows/test-database.yml

```diff
@@ -0,0 +1,185 @@
+      - "main"
+      - "rc"
+      - "hotfix-rc"
+    paths:
+      - ".github/workflows/infrastructure-tests.yml" # This file
+      - "src/Sql/**" # SQL Server Database Changes
+      - "util/Migrator/**" # New SQL Server Migrations
+      - "util/MySqlMigrations/**" # Changes to MySQL
+      - "util/PostgresMigrations/**" # Changes to Postgres
+      - "util/SqliteMigrations/**" # Changes to Sqlite
+      - "src/Infrastructure.Dapper/**" # Changes to SQL Server Dapper Repository Layer
+      - "src/Infrastructure.EntityFramework/**" # Changes to Entity Framework Repository Layer
+      - "test/Infrastructure.IntegrationTest/**" # Any changes to the tests
+  pull_request:
+    paths:
+      - ".github/workflows/infrastructure-tests.yml" # This file
+      - "src/Sql/**" # SQL Server Database Changes
+      - "util/Migrator/**" # New SQL Server Migrations
+      - "util/MySqlMigrations/**" # Changes to MySQL
+      - "util/PostgresMigrations/**" # Changes to Postgres
+      - "util/SqliteMigrations/**" # Changes to Sqlite
+      - "src/Infrastructure.Dapper/**" # Changes to SQL Server Dapper Repository Layer
+      - "src/Infrastructure.EntityFramework/**" # Changes to Entity Framework Repository Layer
+      - "test/Infrastructure.IntegrationTest/**" # Any changes to the tests
+
+jobs:
+  test:
+    name: Run tests
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
+
+      - name: Set up .NET
+        uses: actions/setup-dotnet@3447fd6a9f9e57506b15f895c5b76d3b197dc7c2 # v3.2.0
+
+      - name: Restore tools
+        run: dotnet tool restore
+
+      - name: Docker Compose databases
+        working-directory: "dev"
+        # We could think about not using profiles and pulling images directly to cover multiple versions
+        run: |
+          cp .env.example .env
+          docker compose --profile mssql --profile postgres --profile mysql up -d
+        shell: pwsh
+
+      # I've seen the SQL Server container not be ready for commands right after starting up and just needing a bit longer to be ready
+      - name: Sleep
+        run: sleep 15s
+
+      - name: Migrate SQL Server
+        working-directory: "dev"
+        run: "pwsh ./migrate.ps1"
+        shell: pwsh
+
+      - name: Migrate MySQL
+        working-directory: "util/MySqlMigrations"
+        run: 'dotnet ef database update --connection "$CONN_STR" -- --GlobalSettings:MySql:ConnectionString="$CONN_STR"'
+        env:
+          CONN_STR: "server=localhost;uid=root;pwd=SET_A_PASSWORD_HERE_123;database=vault_dev;Allow User Variables=true"
+
+      - name: Migrate Postgres
+        working-directory: "util/PostgresMigrations"
+        run: 'dotnet ef database update --connection "$CONN_STR" -- --GlobalSettings:PostgreSql:ConnectionString="$CONN_STR"'
+        env:
+          CONN_STR: "Host=localhost;Username=postgres;Password=SET_A_PASSWORD_HERE_123;Database=vault_dev"
```

Comment:
⚠️ The passwords here should be parameterized so that you're not duplicating connection information with the test step. (I assume they don't need to be secret, but treating them as if they were would be nice, too.)